src/app/visa/c.ts

```diff
@@ -0,0 +1,257 @@
+                points += highestDegree.points
+                matches.push(highestDegree)
+            }
+            if (bonus) {
+                points += bonus.points
+                matches.push(bonus)
+            }
+            return { matches, points }
+        }
+    },
+    CAREER: {
+        criteria: [
+            { id: '10_years_or_more', points: 25, match: exp => exp >= 10 },
+            { id: '7_years_or_more', points: 20, match: exp => exp >= 7 },
+            { id: '5_years_or_more', points: 15, match: exp => exp >= 5 },
+            { id: '3_years_or_more', points: 10, match: exp => exp >= 3 },
+        ],
+        match: (criteria, qualifications) => {
+            const match = qualifications.find(q => q.category === 'CAREER') as
+                | CareerQualification
+                | undefined
+            const yearsOfExperience = match?.yearsOfExperience ?? 0
+            return matchMaxPoints(criteria, yearsOfExperience)
+        }
+    },
+    ANNUAL_SALARY: {
+        criteria: [
+            {
+                id: '30m_or_more',
+                points: 50,
+                match: ({ salary }) => salary >= 30_000_000,
+            },
+            {
+                id: '25m_or_more',
+                points: 40,
+                match: ({ salary }) => salary >= 25_000_000,
+            },
+            {
+                id: '20m_or_more',
+                points: 30,
+                match: ({ salary }) => salary >= 20_000_000,
+            },
+            {
+                id: '15m_or_more',
+                points: 20,
+                match: ({ salary }) => salary >= 15_000_000,
+            },
+            {
+                id: '10m_or_more',
+                points: 10,
+                match: ({ salary }) => salary >= 10_000_000,
+            },
+        ],
+        match: (criteria, qualifications) => {
+            const matchSalary = qualifications.find(q => q.category === 'ANNUAL_SALARY') as
+                | AnnualSalaryQualification
+                | undefined
+
+            if (matchSalary === undefined) {
+                return { matches: [], points: 0 }
+            }
+            const salary = matchSalary.salary
+            if (salary < 3_000_000) {
+                throw new Error(errorMessages.salaryTooLow)
+            }
+            return matchMaxPoints(criteria, { salary })
+        }
+    },
+    POSITION: {
+        criteria: [
+            {
+                id: 'representative_director',
+                points: 10,
+            },
+            {
+                id: 'executive_officer',
+                points: 5,
+            }
+
+        ],
+        match: (criteria, allQualifications) => {
+            const qualifications = allQualifications
+                .filter(q => q.category === 'POSITION')
+                .map(q => q.id)
+
+            const matches = criteria.filter(c => qualifications.includes(c.id))
+            const points = matches.reduce(
```

Comment:
Maybe we can extract out this reduce to a function that adds all points, we do it quite often.